af/analysis/analysis/analysis.py

```diff
@@ -535,6 +555,100 @@ def summarize_core_density():
     metrics.gauge("coverage_1_day_5dp", c1)
 
 
+@metrics.timer("plot_msmt_count_per_platform_over_time")
+def plot_msmt_count_per_platform_over_time(conn):
+    log.info("COV: plot_msmt_count_per_platform_over_time")
+    sql = """
+        SELECT date_trunc('day', measurement_start_time) AS day, platform, COUNT(*) AS msm_count
+        FROM fastpath
+        WHERE measurement_start_time >= CURRENT_DATE - interval '60 days'
+            AND measurement_start_time < CURRENT_DATE
+        GROUP BY day, platform
+        ORDER BY day, platform;
+    """
+    q = pd.read_sql_query(sql, conn)
+    p = q.pivot_table(index="day", columns="platform", values="msm_count", fill_value=0)
+    gen_plot("msmt_count_per_platform_over_time", p)
+
+
+@metrics.timer("plot_coverage_per_platform")
+def plot_coverage_per_platform(conn):
+    """Measure how much each platform contributes to measurements
+    """
+    log.info("COV: plot_coverage_per_platform")
+    # Consider only inputs that are listed on citizenlab
+    sql = "SELECT UPPER(cc), COUNT(*) from citizenlab GROUP BY cc"
+    with conn.cursor() as cur:
+        cur.execute(sql)
+        baseline = dict(cur.fetchall())  # CC -> count
+    zz_cnt = baseline.pop("ZZ")
+    for cc in baseline:
+        baseline[cc] += zz_cnt
+
+    # The inner query returns *one* line for each (platform, probe_cc, input)
+    # that has 1 or more msmt. If an input is tested more than once in the time
+    # period in a given CC we treat it as 1.
+    sql = """
+        SELECT platform, probe_cc, count(*)
+        FROM (
+          SELECT platform, probe_cc
+          FROM fastpath
+          WHERE (
+              (probe_cc, input) IN (SELECT UPPER(cc), url FROM citizenlab)
+              OR
+              input IN (SELECT url FROM citizenlab WHERE cc = 'ZZ')
+            )
+            AND measurement_start_time > NOW() - interval '1 days'
```

Comment:
Since the coverage_generator is run every 24h and it's applying the calculation on a 1 day interval from the current time, depending on when it's started for the first time it might run before the batch pipeline has processed the daily batch.

Perhaps the calculation should be done on a 24h window which is 1 day in the past?

Reply:
I plan to change the run frequency in future while keeping the same daily window to monitor improvements when we deploy the URL prioritization.  NOW() makes it flexible.